### PR TITLE
Don’t have Scripts in Modules

### PR DESCRIPTION
**Pull request by V02460**
*This exists because rpmlint complained while packaging Synapse for Fedora.*

This PR removes the shebang lines from all files belonging to Python modules.

Afaict this does not remove functionality:
- For the files under `_script` the shebang can be removed as they are wrapped by a *real* script (with shebang).
- The files under `app` are actually meant to be called directly, but should not occur as cli tools. Without a shebang, they can still be called by their module name, e.g. `python3 -m synapse.app.homeserver`.

Tangential Suggestion: After reading into Python and scripts, using `entry_point` in `setup.py` would make those wrappers scripts unnecessary.

### Pull Request Checklist

* [x] Pull request is based on the develop branch
* [x] Pull request includes a [changelog file](https://github.com/matrix-org/synapse/blob/master/CONTRIBUTING.md#changelog). The entry should:
  - Be a short description of your change which makes sense to users. "Fixed a bug that prevented receiving messages from other servers." instead of "Moved X method from `EventStore` to `EventWorkerStore`.".
  - Use markdown where necessary, mostly for `code blocks`.
  - End with either a period (.) or an exclamation mark (!).
  - Start with a capital letter.
* [x] Pull request includes a [sign off](https://github.com/matrix-org/synapse/blob/master/CONTRIBUTING.md#sign-off)
* [x] Code style is correct (run the [linters](https://github.com/matrix-org/synapse/blob/master/CONTRIBUTING.md#code-style))

### CODE REVIEW

**Comment by babolivier** [comment]
> This exists because rpmlint complained while packaging Synapse for Fedora.

Is passing rpmlint a required step when packaging for Fedora? If not I'd be in favour of keeping these around since they're not doing any harm.

**Comment by V02460** [comment]
Yes, passing rpmlint is part of the [Fedora Packaging Guidelines](https://docs.fedoraproject.org/en-US/packaging-guidelines/#_use_rpmlint) and when run it errors for the given files. It would be preferable to keep downstream changes to a minimum and this would help.

File: synapse/app/admin_cmd.py [hunk 1]
```diff
@@ -1,4 +1,3 @@
-#!/usr/bin/env python
```
Comment (richvdh) [inline synapse/app/admin_cmd.py:1]: looks like you're setting the `+x` bit on this? That seems to be in opposition to the removal of the shebang?

**Review by richvdh — COMMENTED** [review]
looks good otherwise

File: synapse/app/admin_cmd.py (hunk 1 above)
Comment (V02460) [reply synapse/app/admin_cmd.py:1]: Good spot, that wasn’t my intention. Fixed it.

**Review by richvdh — APPROVED** [review]
lgtm, thank you for the contribution!

For future reference: please don't rebase after a review has taken place - it makes it hard to see what has changed since the previous review.

**Pull request merged**